inc/saiicmpecho.h

```diff
@@ -0,0 +1,387 @@
+
+    /**
+     * @brief Destination Port
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_PORT
+     * @condition SAI_ICMP_ECHO_SESSION_ATTR_HW_LOOKUP_VALID == false
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_PORT,
+
+    /**
+     * @brief Source Port
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_PORT
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_RX_PORT,
+
+    /**
+     * @brief Session Global Unique Identifier
+     *
+     * @type sai_uint64_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_GUID,
+
+    /**
+     * @brief Session Cookie
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_COOKIE,
+
+    /**
+     * @brief IP header version
+     *
+     * @type sai_uint8_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_IPHDR_VERSION,
+
+    /**
+     * @brief IP header TOS
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_TOS,
+
+    /**
+     * @brief IP header TTL
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 255
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_TTL,
+
+    /**
+     * @brief Source IP
+     *
+     * @type sai_ip_address_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_SRC_IP_ADDRESS,
+
+    /**
+     * @brief Destination IP
+     *
+     * @type sai_ip_address_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_DST_IP_ADDRESS,
+
+    /**
+     * @brief L2 source MAC address
+     *
+     * @type sai_mac_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @condition SAI_ICMP_ECHO_SESSION_ATTR_HW_LOOKUP_VALID == false
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_SRC_MAC_ADDRESS,
+
+    /**
+     * @brief L2 destination MAC address
+     *
+     * @type sai_mac_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @condition SAI_ICMP_ECHO_SESSION_ATTR_HW_LOOKUP_VALID == false
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_DST_MAC_ADDRESS,
+
+    /**
+     * @brief Transmit interval in microseconds
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_TX_INTERVAL,
+
+    /**
+     * @brief Receive interval in microseconds
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_RX_INTERVAL,
+
+    /**
+     * @brief To enable protection group switch over on session state change
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_ICMP_ECHO_SESSION_ATTR_SET_NEXT_HOP_GROUP_SWITCHOVER,
```

Comment:
This is very fuzzy. What happens if there are multiple such protection NHGs. How does this attribute indicates which one to of the protection NHG takes effect. We just can't trigger all the protection NHG to take over.

Reply:
@chikkaiah-work Please look at this comment as well

Reply:
You brought up a valid point. If it's possible for various NHGs to share the same monitor object, it seems reasonable to consider modifying this attribute to the NHG object ID. Should this attribute be left unset (which is null as default), automatic switchover will not occur. Instead, the switchover should be initiated based on the specified NHG. Please let me know if this sounds fine to you

Reply:
if there are new NHGs added, will the new NHGs be modified based on the monitor session?

Reply:
@lolyu, you are right. Also I have updated the FRR document by mentioning echo session could be included as monitor object.  I have added an example under the section "ICMP echo session with protection group"

Reply:
Thanks for updating the spec with an example workflow. Looks good to me.